storage/entities/base_entity.js

```diff
@@ -0,0 +1,262 @@
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+'use strict';
+
+const { ImplementationPendingError } = require('../errors');
+const filterTypes = require('../utils/filter_types');
+const Field = require('../utils/field');
+const { filterGenerator } = require('../utils/filters');
+
+class BaseEntity {
+	constructor() {
+		this.fields = {};
+		this.filters = {};
+		this.defaultOptions = {
+			limit: 10,
+			offset: 0,
+		};
+		this.defaultFilters = {};
+	}
+
+	/**
+	 * Get one object from persistence layer
+	 *
+	 * @param {string | Object} filters - Multiple filters or just primary key
+	 * @param {Object} options - Extended options
+	 * @param {string} options.fieldSet - Fieldset to get
+	 * @param {Object} tx - transaction object
+	 *
+	 * @return {Promise}
+	 */
+	// eslint-disable-next-line class-methods-use-this,no-unused-vars
```

Comment:
I'm not so sure about disabling these rules in that many places. @nazarhussain could you explain both cases? 

`class-methods-use-this` -> Why not using `static` as written in the https://eslint.org/docs/rules/class-methods-use-this#enforce-that-class-methods-utilize-this-class-methods-use-this ?
`no-unused-vars` -> This rule belongs to the recommended Eslint list. I don't like to disable it at all.

Reply:
@diego-G I think this is ok, as this class is more like a template.
In the extended version of this class, it would those eslint will be removed

Reply:
@diego-G This class is acting as an interface to be implemented by entities.

`class-methods-use-this` -> the method is not a static method and its implementation will contain `this` (thee the Account entity)

`no-unused-vars` -> as there is no implementation here the vars are used as implementation references.

Reply:
I've seen it in many other places not only in this file. It has to be a real exception plus perhaps a comment explaining why. Please, read the rules again and try to use it as less as possible.

Reply:
If it's not an interface I agree it should not be used. Can you point out the other ones you've noticed, pls?